manager/controlapi/secret.go

```diff
@@ -51,6 +51,49 @@ func (s *Server) GetSecret(ctx context.Context, request *api.GetSecretRequest) (
 	return &api.GetSecretResponse{Secret: secret}, nil
 }
 
+// UpdateSecret updates a Secret referenced by SecretID with the given SecretSpec.
+// - Returns `NotFound` if the Secret is not found.
+// - Returns `InvalidArgument` if the SecretSpec is malformed or anything other than Labels is changed
+// - Returns an error if the update fails.
+func (s *Server) UpdateSecret(ctx context.Context, request *api.UpdateSecretRequest) (*api.UpdateSecretResponse, error) {
+	if request.SecretID == "" || request.SecretVersion == nil {
+		return nil, grpc.Errorf(codes.InvalidArgument, errInvalidArgument.Error())
+	}
+
+	var secret *api.Secret
+	err := s.store.Update(func(tx store.Tx) error {
+		secret = store.GetSecret(tx, request.SecretID)
+		if secret == nil {
+			return nil
+		}
+
+		if secret.Spec.Annotations.Name != request.Spec.Annotations.Name {
+			return grpc.Errorf(codes.InvalidArgument, "only updates to Labels are allowed")
```

Comment:
We may allow other fields to be updated in the future.

"renaming secrets is not supported"? That would match the error in #1606.